app/controllers/notifications_controller.rb

```diff
@@ -18,24 +18,42 @@ def index
 
     guardian.ensure_can_see_notifications!(user)
 
+    if notification_types = params[:filter_by_types]&.split(",")&.map(&:to_sym).presence
+      notification_types.map! do |type|
+        if !Notification.types.key?(type)
+          raise Discourse::InvalidParameters.new("invalid notification type: #{type}")
+        end
+        Notification.types[type]
+      end
+    end
+
     if params[:recent].present?
       limit = (params[:limit] || 15).to_i
       limit = 50 if limit > 50
 
-      notifications = Notification.recent_report(current_user, limit)
-      changed = false
-
-      if notifications.present? && !(params.has_key?(:silent) || @readonly_mode)
-        # ordering can be off due to PMs
-        max_id = notifications.map(&:id).max
-        changed = current_user.saw_notification_id(max_id)
+      if notification_types
+        notifications = Notification.recent_report(current_user, limit, notification_types)
+        render_json_dump(
+          notifications: serialize_data(notifications, NotificationSerializer)
+        )
```

Comment:
Now that we're using `Notification.recent_report`, is there a reason for the conditional here?